test/nodl_to_policy/test_policy.py

```diff
@@ -0,0 +1,330 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from lxml import etree
+import nodl
+import nodl._parsing
+import nodl_to_policy.policy as policy
+import pytest
+import sros2
+
+
+def test_init_policy():
+    """Test that an initial empty policy is well-formed."""
+    test_policy = policy.init_policy()
+    assert test_policy.tag == 'policy'
+    assert test_policy.attrib['version'] == sros2.policy.POLICY_VERSION
+    assert len(test_policy) == 1
+    assert test_policy[0].tag == 'enclaves'
+    assert len(test_policy[0]) == 0
+
+
+def test_get_profile_minimal():
+    """Test an XML tree with no pre-existing `enclaves`/`profiles`/`profile` tags."""
+    test_policy = policy.init_policy()
+    test_profile = policy.get_profile(test_policy, node_name='foo')
+    # empty tags get generated after `get_profile` is called
+    test_enclaves = test_policy[0]
+    test_enclave = test_enclaves[0]
+    test_profiles = test_enclave[0]
+
+    # check that a single enclave tag was created
+    assert len(test_enclaves) == 1
+    assert test_enclave.tag == 'enclave'
+    assert test_enclave.attrib['path'] == '/'
+
+    # check that a single profiles tag was created
+    assert len(test_enclave) == 1
+    assert test_profiles.tag == 'profiles'
+
+    # check that a single profile tag was created
+    assert len(test_profiles) == 1
+    assert test_profiles[0] == test_profile
+    test_profile.tag == 'profile'
+    test_profile.attrib['ns'] == '/'
+    test_profile.attrib['node'] == 'foo'
```

Comment:
These are currently "dead" statements; they have no affect.  Should these be `assert`ed?

Reply:
Yes, these should have been assert statements. Fixed in b8a958c.